src/sentry/api/endpoints/team_project_index.py

```diff
@@ -136,3 +148,22 @@ def post(self, request, team):
 
             return Response(serialize(project, request.user), status=201)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def put(self, request, team):
+        if not request.user.is_authenticated():
```

Comment:
you shouldn't need this as the API handles verification of permissions/scopes in the base class you're inheriting (via a permission_classes config)